link-grammar/tokenize/tokenize.c

```diff
@@ -1561,23 +1581,25 @@ static bool mprefix_split(Sentence sent, Gword *unsplit_word, const char *word)
 static bool is_capitalizable(const Dictionary dict, const Gword *word)
 {
 	/* Words at the start of sentences are capitalizable */
-	if (MT_WALL == word->prev[0]->morpheme_type) return true;
-	if (MT_INFRASTRUCTURE == word->prev[0]->morpheme_type) return true;
+	if (MT_WALL == (*gwordlist_at(word->prev, 0))->morpheme_type)
+		return true;
+	if (MT_INFRASTRUCTURE == (*gwordlist_at(word->prev, 0))->morpheme_type)
+		return true;
 
 	/* Words following colons are capitalizable. */
 	/* Mid-text periods and question marks are sentence-splitters. */
-	if (strcmp(":", word->prev[0]->subword) == 0 ||
-		 strcmp(".", word->prev[0]->subword) == 0 ||
-		 strcmp("...", word->prev[0]->subword) == 0 ||
-		 strcmp("…", word->prev[0]->subword) == 0 ||
-		 strcmp("?", word->prev[0]->subword) == 0 ||
-		 strcmp("!", word->prev[0]->subword) == 0 ||
-		 strcmp("？", word->prev[0]->subword) == 0 ||
-		 strcmp("！", word->prev[0]->subword) == 0 )
+	if (strcmp(":", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp(".", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp("...", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp("…", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp("?", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp("!", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp("？", (*gwordlist_at(word->prev, 0))->subword) == 0 ||
+		 strcmp("！",(*gwordlist_at(word->prev, 0))->subword) == 0)
 		return true;
```

Comment:
!!!

Reply:
In any case (even if we don't apply this PR) this can be replaced by checking an affix class, something like:
`":.…?!？！" "...": AFEND+; % 2 tokens`
When the first component is checked by `in_afdict_class()` and anything afterwards (here we have only one token afterwards) by a loop. This also makes `AFDICT_BULLETS ` more general since it is then possible to to easily add bullets longer than one characters.

`